packages/experiments/src/components/Shimmer/Shimmer.base.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  classNamesFunction,
+} from '../../Utilities';
+import {
+  IShimmerProps,
+  IShimmerStyleProps,
+  IShimmerStyles,
+  ShimmerElementType,
+  ICircle,
+  ILine,
+  IGap,
+  ShimmerElementVerticalAlign,
+} from './Shimmer.types';
+import {
+  DefaultPalette,
+  IStyleSet
+} from '../../Styling';
+import { ShimmerRectangle } from './ShimmerRectangle/ShimmerRectangle';
+import { ShimmerCircle } from './ShimmerCircle/ShimmerCircle';
+
+const getClassNames = classNamesFunction<IShimmerStyleProps, IShimmerStyles>();
+
+export class ShimmerBase extends BaseComponent<IShimmerProps, {}> {
+  public static defaultProps: IShimmerProps = {
+    width: 100,
+    isDataLoaded: false
+  };
+  private _classNames: {[key in keyof IShimmerStyles]: string};
+  constructor(props: IShimmerProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const { getStyles, width, lineElements, children, isDataLoaded } = this.props;
+    const maxHeight: number | undefined = lineElements ? this._findMaxHeight(lineElements) : undefined;
+    this._classNames = getClassNames(getStyles!, { width, maxHeight, isDataLoaded });
+
+    const elements: JSX.Element[] | JSX.Element = lineElements ?
+      lineElements.map((elem: ICircle | ILine | IGap, index: number): JSX.Element => {
+        switch (elem.type) {
+          case ShimmerElementType.CIRCLE:
+            return (
+              <ShimmerCircle
+                key={ index }
+                { ...elem }
+                borderAlignStyle={ this._getBorderAlignStyles(maxHeight, elem) }
+              />
+            );
+          case ShimmerElementType.GAP:
+            const gapWidth = elem.width ? elem.width + '%' : '1%';
```

Comment:
Doesn't 1% default seem really less? What do we expect the most probable value here?

Reply:
Corrected this one and defaulted to 5px due to having now two options for width in pixels and percentages.